### PR TITLE
zbus: isolated msg subscriber pools by channel 

### PR DESCRIPTION
**Pull request by rodrigopex**
Currently, zbus uses a single global `net_buf` pool to publish messages to msg_subscribers. It would be good to have a way to separate the pools for channels related to critical parts of the systems to avoid publication failure on these particular channels. These channels will not use the global pool. They can set an isolated pool by calling the `zbus_chan_set_msg_sub_pool.`

### CODE REVIEW

File: subsys/zbus/Kconfig
Comment (ubieda) [inline subsys/zbus/Kconfig:42]:
```suggestion
	bool "Enable isolated pool instead of only using the global pool."
```

File: samples/subsys/zbus/msg_subscriber/src/main.c
Comment (ubieda) [inline samples/subsys/zbus/msg_subscriber/src/main.c:202]: I'm also curious whether this is a non-related change

File: samples/subsys/zbus/msg_subscriber/prj.conf
Comment (ubieda) [inline samples/subsys/zbus/msg_subscriber/prj.conf:5]: I'm curious on the rationale behind bumping up the priority on this sample. Is it a related change?

**Comment by ubieda** [comment]
This PR looks good to me! Just left a couple comments and a suggestion.

File: samples/subsys/zbus/msg_subscriber/src/main.c
Comment (rodrigopex) [reply samples/subsys/zbus/msg_subscriber/src/main.c:202]: Removed.

File: subsys/zbus/Kconfig
Comment (rodrigopex) [reply subsys/zbus/Kconfig:42]: Done. I have used the text: "Use isolated pools instead of only using the global pool." based on the rule of not using "enable" for prompts: https://docs.zephyrproject.org/latest/build/kconfig/tips.html#prompt-strings

**Review by rodrigopex — COMMENTED** [review]
@ubieda, unrelated changes removed and the requested changes attended.

**Review by ubieda — DISMISSED** [review]

File: samples/subsys/zbus/msg_subscriber/prj.conf
Comment (rodrigopex) [reply samples/subsys/zbus/msg_subscriber/prj.conf:5]: This was necessary to ensure the correct sequence of outputs and to cover the code with the sample working as a test.

**Review by ubieda — APPROVED** [review]

**Review by bjarki-andreasen — APPROVED** [review]

**Comment by aescolar** [comment]
Bulk changing the milestone from everything which did not make it to the v3.7.0 freeze deadline
and which is not labeled as a bug or documentation (or does not appear clearly as only containing
a bug fix or a documentation change).
If you disagree about this change please add the milestone again.

Please note that until rc2 only PRs containing bug fixes, docs and tests can be merged.
After that and until release only bug fixes and docs changes.

Exceptions require approval by the release team and a vote by the TSC.
https://docs.zephyrproject.org/latest/project/release_process.html#stabilization-phase

**Pull request merged**
